GameServerLib/Items/Shop.cs

```diff
@@ -74,13 +73,18 @@ public bool HandleItemBuyRequest(int itemId)
             return true;
         }
 
-        private void RemoveItem(IItem item, byte slotId, byte stackSize = 0)
+        private void RemoveItem(IItem item, byte slotId, byte newStacks)
```

Comment:
This class is actually need major rework. The remove item method should be in champion IObjBase class instead of here, and Shop shouldn't have `_owner` field. Basically, there should be 2 shops (one for each team) in our case.
This part:
```
            var inventory = _owner.Inventory;
            _game.PacketNotifier.NotifyRemoveItem(_owner, slotId, newStacks);

            if(newStacks == 0) // I don't believe there exists a stackable item that applies stats.
            {
                _owner.Stats.RemoveModifier(item.ItemData);
                _owner.RemoveSpell((byte)(slotId + ITEM_ACTIVE_OFFSET));
                inventory.RemoveItem(item);
            }
``` 
is actually should be part of IObjBase class.